pkg/endpoint/connector/docker.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2016-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	// ContainerInterfacePrefix is the container's internal interface name prefix.
+	ContainerInterfacePrefix = "cilium"
+)
+
+// DockerNetNSConfigurer is a wrapper around the configuration of a network
+// namespace for IPVLAN integration.
+type DockerNetNSConfigurer struct{}
+
+// ConfigureNetNSForIPVLAN sets up IPVLAN in the specified network namespace.
+// Returns the file descriptor for the tail call map / ID, and an error if
+// any operation while configuring said namespace fails.
+func (d *DockerNetNSConfigurer) ConfigureNetNSForIPVLAN(netNsPath string) (mapFD, mapID int, err error) {
```

Comment:
We already have a `pkg/endpoint/ipvlan.go` that seems to have same-ish enough functions for this to go in there. I'm not sure how docker-specific this function is and if you wanted to highlight that part more clearly (perhaps you have more docker-specific things in mind for later?)

Reply:
Agreed, function will be moved to `ipvlan.go`.